rcl/test/rcl/test_events.cpp

```diff
@@ -366,21 +410,26 @@ TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_pubsub_no_deadline_
     EXPECT_EQ(std::string(test_string), std::string(msg.string_value.data, msg.string_value.size));
   }
 
-  ASSERT_TRUE(subscription_event_ready);
-  if (subscription_event_ready) {
+  // test subscriber/datareader deadline missed status
+  EXPECT_FALSE(subscription_event_ready);
+  {
     rmw_requested_deadline_missed_status_t deadline_status;
     ret = rcl_take_event(&subscription_event, &deadline_status);
     ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
     EXPECT_EQ(deadline_status.total_count, 0);
     EXPECT_EQ(deadline_status.total_count_change, 0);
   }
 
-  ASSERT_TRUE(publisher_event_ready);
-  if (publisher_event_ready) {
+  // test publisher/datawriter deadline missed status
+  EXPECT_FALSE(publisher_event_ready);
+  {
     rmw_offered_deadline_missed_status_t deadline_status;
     ret = rcl_take_event(&publisher_event, &deadline_status);
     ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
     EXPECT_EQ(deadline_status.total_count, 0);
     EXPECT_EQ(deadline_status.total_count_change, 0);
   }
+
+  // clean up
+  TearDownPubSub();
```

Comment:
shouldn't you be able to do auto teardown using the class fixture?

Reply:
For the liveliness test, I can only tear down the subscriber, so I couldn't make the auto `TearDown()` tear down absolutely everything.

Reply:
Hmm, I guess it's already doing a manual setup, so it's not out of pattern. 

You could perhaps on the fixture set a `this->doTearDownPublisher` and check that on the TearDown method - might keep more logic contained on the fixture such that you can't accidentally forget to clean up

Not blocking approval on this, just picking through it